src/func_gpt.py

```diff
@@ -74,29 +56,36 @@ def chat(self, message):
             func = self.funcs[func_name]
             func_result = func(args)
 
-            self.context.append({"role": "function", "name": func_name, "content": func_result})
-
-            try:
-                follow_up = self.oai.chat.completions.create(
-                    messages=self.context,
-                    model=self.model,
-                    tools=self.funcs_desc,
-                    tool_choice="auto"
-                )
-                logger.info(f"Follow-up response: {follow_up}")
-                if not follow_up.choices or not follow_up.choices[0].message.content:
-                    raise ValueError("Follow-up API response is empty or invalid")
-            except Exception as e:
-                logger.error(f"Follow-up API call failed: {e}")
+            if func_result and isinstance(func_result, str):
+                self.context.append({"role": "function", "name": func_name, "content": func_result})
+            else:
+                logger.error(f"Invalid function result for {func_name}: {func_result}")
+                return "Invalid function result."
+
+            follow_up = self.oai.chat.completions.create(
+                messages=self.context,
+                model=self.model,
+                tools=self.funcs_desc,
+                tool_choice="auto"
+            )
 
             final_output = follow_up.choices[0].message.content
-            self.context.append({"role": "assistant", "content": final_output})
+            if final_output and isinstance(final_output, str):
+                self.context.append({"role": "assistant", "content": final_output})
+            else:
+                logger.error("Received invalid response from assistant.")
```

Comment:
```suggestion
                logger.error("FuncGPT: Received invalid response from assistant.")
```
но по такому сообщению в логах тоже мало будет понятно, без контекста